invisible_cities/detector_simulation/buffer_functions.py

```diff
@@ -0,0 +1,149 @@
+    bin_sensors = sensors.groupby('sensor_id').apply(weighted_histogram    ,
+                                                     np.append(bins, b_pad))
+    return bins, bin_sensors
+
+
+## !! to-do: clarify for non-pmt versions of next
+## !! to-do: Check on integral instead of only threshold?
+def find_signal(wfs          : pd.Series,
+                bin_threshold: float    ,
+                stand_off    : int      ) -> List[int]:
+    """
+    Finds signal in the binned waveforms and
+    identifies candidate triggers.
+    """
+    eng_sum = wfs.sum()
+    indices = indices_and_wf_above_threshold(eng_sum,
+                                             bin_threshold).indices
+    ## Just using this and the stand_off for now
+    ## taking first above sum threshold.
+    ## !! To-do: make more robust with min int? or similar
+    all_indx = split_in_peaks(indices, stand_off)
+    return [pulse[0] for pulse in all_indx]
+
+
+def padder(sensors: np.ndarray, padding: Tuple) -> np.ndarray:
+    """Pads zeros around each sensor in a 2D array"""
+    if not sensors.shape[0]:
+        return np.empty((0, padding[0] + padding[1] + 1))
+    return np.apply_along_axis(np.pad, 1, sensors, padding, "constant")
+
+
+def calculate_buffers(buffer_len: float, pre_trigger: float,
+                      pmt_binwid: float, sipm_binwid: float) -> Callable:
+    """
+    Calculates the output buffers for all sensors
+    based on a configured buffer length and pretrigger.
+    Synchronising the clock between the two sensor types.
+
+    Parameters
+    ----------
+    buffer_len  : float
+                  Length of buffer expected in mus
+    pre_trigger : float
+                  Time in buffer before identified signal in mus
+    pmt_binwid  : float
+                  Width in mus of PMT sample integration
+    sipm_binwid : float
+                  Width in mus of SiPM sample integration
+    """
+    pmt_buffer_samples  = int(buffer_len * units.mus /  pmt_binwid)
+    sipm_buffer_samples = int(buffer_len * units.mus / sipm_binwid)
+    sipm_pretrg         = int(pre_trigger * units.mus / sipm_binwid)
+    sipm_postrg         = sipm_buffer_samples - sipm_pretrg
+    pmt_pretrg_base     = int(pre_trigger * units.mus / pmt_binwid)
+    pmt_postrg_base     = pmt_buffer_samples - pmt_pretrg_base
+
+    def generate_slice(trigger    : List      ,
+                       pmt_bins   : np.ndarray,
+                       pmt_charge : np.ndarray,
+                       sipm_bins  : np.ndarray,
+                       sipm_charge: np.ndarray) -> Tuple:
+        """
+        Synchronises the clocks between the SiPMs and PMTs
+        and slices the histograms where this synchronisation
+        indicates.
+        """
+        npmt_bin  = len(pmt_bins)
+        nsipm_bin = len(sipm_bins)
+
+        trg_bin    = np.where(sipm_bins <= pmt_bins[trigger])[0][-1]
+        bin_corr   = (pmt_bins[trigger] - sipm_bins[trg_bin]) / pmt_binwid
```

Comment:
```suggestion
        bin_corr   = (pmt_bins[trigger] - sipm_bins[trg_bin]) // pmt_binwid
```
It doesn't change anything, cause there is an `int` truncation just after this line, but it is more correct algorithmically.